pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -473,6 +473,12 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 		}
 	}
 	tlsInspector := func() *listener.ListenerFilter {
+		// If we have enrolled workloads, we should sniff TLS
+		// This is important if users want to assert authz against SNI.
+		if len(wls) > 0 {
+			// Default inspector with no excluded ports, workloads could listen on any port
+			return xdsfilters.TLSInspector
+		}
```

Comment:
sorry for the somewhat irrelevant question, but is there any way `tlsInspector` becomes nil? we have a nil check later on

https://github.com/istio/istio/blob/522bbac962b8f3db4a31c969aca921f1b489b760/pilot/pkg/networking/core/listener_waypoint.go#L539-L541

Reply:
If you have no workloads and there are no `nonInspectorPorts` then we'll return nil and not configure the inspector.

Reply:
ohhh, i didn't realize the function got called.